### PR TITLE
Workaround to fix broken curl installation

### PR DESCRIPTION
**Pull request by hvelab**
As reported in the EESSI support portal issue [#25](https://gitlab.com/eessi/support/-/issues/25), curl fails in RHEL 8 and above systems as CA files location differs. Works for all rhel based systems as I could reproduce the issue also in them.

```
$ cat /etc/os-release 
NAME="Rocky Linux"
VERSION="9.3 (Blue Onyx)"
ID="rocky"
ID_LIKE="rhel centos fedora"

$ source software-layer/init/bash 
Found EESSI repo @ /cvmfs/software.eessi.io/versions/2023.06!
archdetect says x86_64/intel/skylake_avx512
Using x86_64/intel/skylake_avx512 as software subdirectory.
[...]
Environment set up to use EESSI (2023.06), have fun!

{EESSI 2023.06} $ ml CMake
{EESSI 2023.06} $ curl https://www.example.com
<!doctype html>
<html>
<head>
    <title>Example Domain</title>
[...]
```

I am not a fan of hardcording in the init script but investigating the issue I haven't found a more suitable way to solve it for our use case. Maybe another solution would be to add a modlua footer exporting this variable in the affected modules?

### CODE REVIEW

File: init/eessi_environment_variables
Comment (ocaisa) [inline init/eessi_environment_variables:27]: Given that you know the path where the certs should be, isn't it enough to check if it exists and if so export the variable?

**Comment by ocaisa** [comment]
Currently we ship `curl` in the compat layer, so there's no way to avoid setting this in the initialisation script. If it ever came to be that we stopped doing that, then it would be enough to set this via an Lmod hook

File: init/eessi_environment_variables
Comment (ocaisa) [reply init/eessi_environment_variables:27]: No need for an OS check with that approach

Comment (hvelab) [reply init/eessi_environment_variables]: True! I wanted to make sure that nothing breaks, but you are right, my implementation was over-engineered, now its much more simple

**Review by ocaisa — CHANGES_REQUESTED** [review]
Looks good but can you add a comment to mention that if curl we're removed from the compat layer we could do this via the cURL module file (or an Lmod hook) instead

File: init/eessi_environment_variables
Comment (ocaisa) [inline init/eessi_environment_variables:110]:
```suggestion
	# This is required here because we ship curl in our compat layer. If we only provided
	# curl as a module file we could instead do this via a `modluafooter` in an EasyBuild
	# hook (or via an Lmod hook)
	rhel_libcurl_file="/etc/pki/tls/certs/ca-bundle.crt"
```

File: init/eessi_environment_variables
Comment (ocaisa) [inline init/eessi_environment_variables:117]:
```suggestion
          fi

          # Fix wrong path for RHEL >=8 libcurl 
	  # This is required here because we ship curl in our compat layer. If we only provided
	  # curl as a module file we could instead do this via a `modluafooter` in an EasyBuild
	  # hook (or via an Lmod hook)
	  rhel_libcurl_file="/etc/pki/tls/certs/ca-bundle.crt"
          if [ -f $rhel_libcurl_file ]; then
            show_msg "Found libcurl CAs file at RHEL location, setting CURL_CA_BUNDLE"
            export CURL_CA_BUNDLE=$rhel_libcurl_file
          fi
```
indentation was off

**Review by ocaisa — CHANGES_REQUESTED** [review]

**Comment by xinan1911** [comment]
@ocaisa Could you take a final check before merging this PR?

**Comment by ocaisa** [comment]
I can't merge this because I don't know how to deploy it

**Comment by bedroge** [comment]
I'm not completely sure anymore if this can now be done with the bot. Let's try it, and otherwise I can do it manually.

**Comment by bedroge** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/generic

**Comment by bedroge** [comment]
Looks like it actually worked :tada:.

I also tested the fix on our Rocky 8 cluster: I got an error without the fix, but after applying the fix the curl command worked fine.

**Pull request merged**
